src/main/java/thePackmaster/actions/upgradespack/SharpeningFollowUpAction.java

```diff
@@ -0,0 +1,24 @@
+package thePackmaster.actions.upgradespack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.actions.utility.WaitAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+
+public class SharpeningFollowUpAction extends AbstractGameAction {
+
+    public SharpeningFollowUpAction() {
+        duration = 0.01f;
+    }
+
+    @Override
+    public void update() {
+        AbstractDungeon.actionManager.addToTop(new WaitAction(0.2F));// 18
```

Comment:
I don't understand this one? It's not just a wait action. It's a bit weirdly maid out because I copied Scrape's follow up action, but it doesn't just wait. It waits then upgrades the cards

Reply:
Ah, I missed the DrawnCards callback, makes sense.